### PR TITLE
Add test case for AuthorityChecker

### LINKED ISSUE #11257: Add test case for AuthorityChecker
# Aim
Add test case for AuthorityChecker

### PR DESCRIPTION
**Pull request by zhujunxxxxx**
Fixes #11257.

Changes proposed in this pull request:
- Add test case for `AuthorityChecker`

### CODE REVIEW

**Comment by zhujunxxxxx** [comment]
@tristaZero please review.

File: shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/test/java/org/apache/shardingsphere/authority/checker/AuthorityCheckerTest.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        users.add(root);
+        AuthorityRuleConfiguration ruleConfig = new AuthorityRuleConfiguration(Collections.emptyList(), new ShardingSphereAlgorithmConfiguration("NATIVE", new Properties()));
+        AuthorityRule rule = new AuthorityRule(ruleConfig, createMetaDataMap(users), users);
+        SQLChecker<AuthorityRule> sqlChecker = OrderedSPIRegistry.getRegisteredServices(Collections.singleton(rule), SQLChecker.class).get(rule);
+        assertThat(sqlChecker, notNullValue());
+        // true
+        assertThat(sqlChecker.check("db0", new Grantee("root", "localhost"), rule), is(true));
+        // false
+        assertThat(sqlChecker.check("db1", new Grantee("root", "localhost"), rule), is(false));
+        // false
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/test/java/org/apache/shardingsphere/authority/checker/AuthorityCheckerTest.java:92]: I can see these results from assertThat, therefore what's the point of these comments?

File: shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/test/java/org/apache/shardingsphere/authority/checker/AuthorityCheckerTest.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.Silent.class)
+public class AuthorityCheckerTest {
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/test/java/org/apache/shardingsphere/authority/checker/AuthorityCheckerTest.java:57]: Is it better to define it as a final class?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/test/java/org/apache/shardingsphere/authority/checker/AuthorityCheckerTest.java (hunk 1 above)
Comment (zhujunxxxxx) [reply shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/test/java/org/apache/shardingsphere/authority/checker/AuthorityCheckerTest.java:92]: I will remove it

File: shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/test/java/org/apache/shardingsphere/authority/checker/AuthorityCheckerTest.java (hunk 2 above)
Comment (zhujunxxxxx) [reply shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/test/java/org/apache/shardingsphere/authority/checker/AuthorityCheckerTest.java:57]: ok

**Comment by zhujunxxxxx** [comment]
@tristaZero this PR is ready to merge.

**Review by tristaZero — APPROVED** [review]
@zhujunxxxxx Sure, merged.

**Pull request merged**